### PR TITLE
MINOR: add the systemApp property

### PR DESCRIPTION
**Pull request by sushi30**
### Describe your changes:

Add the option for an app to be defined as a system app so it cannot be uninstalled.

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/apps/AppResource.java [hunk 1]
```diff
@@ -686,6 +687,9 @@ public Response delete(
           @PathParam("name")
           String name) {
     App app = repository.getByName(null, name, repository.getFields("bot,pipelines"));
+    if (app.getSystemApp()) {
+      throw new AuthorizationException("System App cannot be uninstalled");
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/resources/apps/AppResource.java:691]: can we not use AuthorizationException here or introduce a new exception.
AuthorizationException should be used specifically for permissions not for anything else.

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/apps/AppResource.java [hunk 2]
```diff
@@ -711,6 +716,9 @@ public Response delete(
       @Parameter(description = "Id of the App", schema = @Schema(type = "UUID")) @PathParam("id")
           UUID id) {
     App app = repository.get(null, id, repository.getFields("bot,pipelines"));
+    if (app.getSystemApp()) {
+      throw new AuthorizationException("System App cannot be uninstalled");
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/resources/apps/AppResource.java:720]: same here

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/apps/AppResource.java (hunk 1 above)
Comment (sushi30) [reply openmetadata-service/src/main/java/org/openmetadata/service/resources/apps/AppResource.java]: changed to BadRequest

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/apps/AppResource.java (hunk 2 above)
Comment (sushi30) [reply openmetadata-service/src/main/java/org/openmetadata/service/resources/apps/AppResource.java]: same

**Review by harshach — APPROVED** [review]

**Pull request merged**